src/components/VHeader/VHeaderInternal.vue

```diff
@@ -0,0 +1,137 @@
+          />
+        </template>
+
+        <template #top-bar>
+          <div
+            class="flex h-20 w-full justify-between bg-dark-charcoal py-4 text-white pe-3 ps-6"
+          >
+            <VBrand :is-fetching="false" />
+            <VIconButton
+              ref="closeButton"
+              variant="plain"
+              :button-props="{ variant: 'plain' }"
+              :icon-props="{ iconPath: closeIcon }"
+              size="search-medium"
+              class="border-tx text-white"
+              :aria-label="$t('modal.close')"
+              @click="setModalVisibility(false)"
+            />
+          </div>
+        </template>
+        <template #default>
+          <ul class="flex flex-col items-end bg-dark-charcoal text-white">
+            <li v-for="page in allPages" :key="page.id" class="mt-6">
+              <VLink
+                class="heading-5 text-white"
+                :class="currentPage === page.id ? 'font-bold' : ''"
+                :href="page.link"
+                show-external-icon
+                @click="setModalVisibility(false)"
+                >{{ $t(`navigation.${page.id}`) }}</VLink
+              >
+            </li>
+          </ul>
+          <VLink
+            href="https://wordpress.org"
+            class="text-white hover:no-underline"
+          >
+            <i18n
+              tag="p"
+              path="footer.wordpress-affiliation"
+              class="mt-auto flex flex-row items-center text-sm"
+            >
+              <template #wordpress>
+                <WordPress class="aria-hidden text-white" />
+                <span class="sr-only">WordPress</span>
+              </template>
+            </i18n>
+          </VLink>
+        </template>
+      </VModal>
+    </nav>
+  </header>
+</template>
+<script lang="ts">
+import { defineComponent, ref, useRoute, watch } from '@nuxtjs/composition-api'
+
+import usePages from '~/composables/use-pages'
+
+import VIconButton from '~/components/VIconButton/VIconButton.vue'
+import VBrand from '~/components/VBrand/VBrand.vue'
+import VLink from '~/components/VLink.vue'
+import VModal from '~/components/VModal/VModal.vue'
+
+import closeIcon from '~/assets/icons/close.svg'
+import menuIcon from '~/assets/icons/menu.svg'
+import WordPress from '~/assets/wordpress.svg?inline'
+
+export default defineComponent({
+  name: 'VHeaderInternal',
+  components: { VBrand, VIconButton, VLink, VModal, WordPress },
+  setup() {
+    const menuButtonRef = ref<InstanceType<typeof VIconButton>>(null)
+
+    const { all: allPages, current: currentPage } = usePages(true)
+    const route = useRoute()
+
+    const isModalVisible = ref(false)
+    const setModalVisibility = (visibility: boolean) => {
+      isModalVisible.value = visibility
+    }
+
+    // When clicking on an internal link in the modal, close the modal
+    watch(route, () => {
+      if (isModalVisible.value) {
+        setModalVisibility(false)
+      }
+    })
```

Comment:
The modal closes when you click an external link as well. Is that intentional?

The modal also causes the page to scroll to the top now instead of staying at the same scroll position. This appears to be a regression in this PR as it does not do it on the staging deployment.

Both behaviours are illustrated in the video below.


https://user-images.githubusercontent.com/24264157/191396993-9fdac266-f8d6-48c9-8855-5bb4540c83ba.mp4

Reply:
> The modal also causes the page to scroll to the top now instead of staying at the same scroll position. This appears to be a regression in this PR as it does not do it on the staging deployment.

Observing this too.

Reply:
I think I've fixed this in https://github.com/WordPress/openverse-frontend/pull/1825/commits/e445d9a14426ee58b9c911aaefbe77e2e8056159, are you still seeing this behavior, @zackkrida ?